runhouse/rns/run.py

```diff
@@ -0,0 +1,560 @@
+            self._convert_symlink_to_file(path=fp)
+
+        if system == "here":
+            # Save to default local path if none provided
+            path = path or self.base_local_folder_path()
+
+        new_run._folder = self._folder.to(
+            system=system, path=path, data_config=data_config
+        )
+
+        return new_run
+
+    def inputs(self) -> bytes:
+        """Read the pickled function inputs saved on the system for the Run.
+        Note: It's the user's responsibility to unpickle the inputs."""
+        inputs_path = self.fn_inputs_path()
+        if not self.exists_in_system(inputs_path):
+            raise FileNotFoundError(
+                f"No inputs found on {self.system} in path: {inputs_path}"
+            )
+
+        return blob(path=inputs_path, system=self.system).fetch()
+
+    def result(self) -> Union[Any, None]:
+        """Read and deserialize the function result saved on the system for the Run.
+        If no result file is found, return None."""
+        result_path = self.fn_result_path()
+        if not self.exists_in_system(result_path):
+            logger.warning(
+                f"No result saved on {self.system} in path: {result_path}. Please re-run, or "
+                f"wait until the run has completed. (current status: {self.status})"
+            )
+            return None
+
+        return pickle.loads(blob(path=result_path, system=self.system).fetch())
+
+    def stdout(self):
+        """Read the stdout saved on the system for the Run."""
+        stdout_path = self._stdout_path
+        logger.info(f"Reading stdout from path: {stdout_path}")
+        if not self.exists_in_system(stdout_path):
+            raise FileNotFoundError(f"No stdout file found in path: {stdout_path}")
+
+        return (
+            blob(path=stdout_path, system=self.system).fetch().decode().strip()
+        )
+
+    def stderr(self):
+        """Read the stderr saved on the system for the Run."""
+        stderr_path = self._stderr_path
+        logger.info(f"Reading stderr from path: {stderr_path}")
+        if not self.exists_in_system(stderr_path):
+            raise FileNotFoundError(f"No stderr file found in path: {stderr_path}")
+
+        return (
+            blob(path=stderr_path, system=self.system).fetch().decode().strip()
+        )
+
+    def delete_in_system(self, path: str = None):
+        """Delete a Run from its system. Deletes the specified folder and all its contents from the Run's file system"""
+        path = path or self.path
+        self._folder.delete_in_system(path)
+
+    def exists_in_system(self, path: str = None) -> bool:
+        """Checks whether a path exists in the Run's file system.
+        Optionally provide a custom path to check, otherwise will default to the Run's folder path."""
+        return self._folder.fsspec_fs.exists(path or self.path)
+
+    def register_new_fn_run(self):
```

Comment:
I think these register functions should be private?

Reply:
we are calling these functions as part of an RPC or http endpoint when we create a new Run object on the server

Reply:
private as in we prob don't expect users to touch this and can preface it with an underscore?